apps/frontend/src/services/semester/SemesterQuery.ts

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from "@tanstack/react-query"
+import { QueryKeys } from "@/services"
+import SemesterService from "./SemesterService"
+
+const SemesterQuery = {
+  /**
+   * Retrieves and caches all semesters.
+   *
+   * @returns A query hook that fetches all semesters
+   */
+  useGetSemesters: () => {
+    return useQuery({
+      queryKey: [QueryKeys.SEMESTER_QUERY_KEY],
+      queryFn: async () => {
+        const response = await SemesterService.getSemesters()
+        return response?.data ?? []
+      },
```

Comment:
you can just use `SemesterService.getSemesters`

```suggestion
      queryFn: SemesterService.getSemesters
```